frontend-runitup/src/Components/Feed/CommentSection.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from "react";
+import { List, Avatar, Form, Input, Button, Typography, Divider, message } from "antd";
+import { SendOutlined } from "@ant-design/icons";
+import { getHeaders, generateColor } from "../../utils/apiConfig";
+import "../../styles/CommentSection.css";
+
+const { TextArea } = Input;
+const { Text } = Typography;
+
+const CommentSection = ({ postId, onCommentAdded }) => {
+  const [comments, setComments] = useState([]);
+  const [submitting, setSubmitting] = useState(false);
+  const [value, setValue] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [hasMore, setHasMore] = useState(true);
+
+  useEffect(() => {
+    fetchComments();
+  }, [postId]);
+
+  const fetchComments = async (offset = 0) => {
+    setLoading(true);
+    try {
+      const response = await fetch(
+        `${
+          import.meta.env.VITE_POST_ADDRESS
+        }/posts/${postId}/comments?offset=${offset}&limit=10`,
+        { headers: getHeaders() }
+      );
+      if (!response.ok) throw new Error("Failed to fetch comments");
+      const data = await response.json();
+      setComments((prevComments) =>
+        offset === 0 ? data : [...prevComments, ...data]
+      );
+      setHasMore(data.length === 10);
+    } catch (error) {
+      message.error("Failed to load comments:", error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleSubmit = async () => {
+    if (!value.trim()) return;
+
+    setSubmitting(true);
+    try {
+      const response = await fetch(
+        `${import.meta.env.VITE_POST_ADDRESS}/posts/${postId}/comments`,
+        {
+          method: "POST",
+          headers: getHeaders(),
+          body: JSON.stringify({ content: value.trim() }),
+        }
+      );
+      if (!response.ok) throw new Error("Failed to add comment");
+      const newComment = await response.json();
+      setComments([newComment, ...comments]);
+      setValue("");
+      if (onCommentAdded) onCommentAdded();
+    } catch (error) {
+      message.error("Failed to add comment:", error);
+    } finally {
+      setSubmitting(false);
+    }
+  };
+
+  const loadMoreComments = () => {
+    fetchComments(comments.length);
```

Comment:
This could be a later thing but consider adding pagination for comments especially if comments thread becomes super long. Just a suggestion though